### PR TITLE
Extract Aspire.Hosting.Dapr.Tests

### PR DESCRIPTION
**Pull request by sebastienros**
Part of https://github.com/dotnet/aspire/issues/4294
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5282)

### CODE REVIEW

File: tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Tests.Schema;
+using Xunit;
+
+namespace Aspire.Hosting.Dapr.Tests;
+
+public class AzureSchemaTests
```
[the remark this answers is not in the document]
Comment (sebastienros) [reply tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs:9]: Typo

File: Aspire.sln [hunk 2]
```diff
@@ -455,9 +455,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Extensions.Servic
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "OracleEndToEnd", "OracleEndToEnd", "{16F4BE51-EEC1-4C72-9148-5289F4BBA38C}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "OracleEndToEnd.AppHost", "playground\OracleEndToEnd\OracleEndToEnd.AppHost\OracleEndToEnd.AppHost.csproj", "{8AC2375F-89F4-4E20-BF2E-F243AB1B2A10}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OracleEndToEnd.AppHost", "playground\OracleEndToEnd\OracleEndToEnd.AppHost\OracleEndToEnd.AppHost.csproj", "{8AC2375F-89F4-4E20-BF2E-F243AB1B2A10}"
```
Comment (radical) [inline Aspire.sln:458]: What exactly is being changed in the sln?

Comment (sebastienros) [reply Aspire.sln:458]: VS automatically updated the project types when I added the new project. This "new" type is the same as before but uses the new project system. At some point we need to accept these changes or they will always appear in random PRs.

**Review by radical — APPROVED** [review]

File: src/Aspire.Hosting/Aspire.Hosting.csproj [hunk 3]
```diff
@@ -68,6 +68,7 @@
   <ItemGroup>
     <InternalsVisibleTo Include="Aspire.Hosting.Tests" />
     <InternalsVisibleTo Include="Aspire.Hosting.Containers.Tests" />
+    <InternalsVisibleTo Include="Aspire.Hosting.Dapr.Tests" />
```
Comment (eerhardt) [inline src/Aspire.Hosting/Aspire.Hosting.csproj:71]: We need to break this. What is this needed for?

File: tests/Aspire.Hosting.Dapr.Tests/Aspire.Hosting.Dapr.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Aspire.Hosting.Dapr\Aspire.Hosting.Dapr.csproj" />
+    <ProjectReference Include="..\..\src\Aspire.Hosting.Testing\Aspire.Hosting.Testing.csproj" />
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Dapr.Tests/Aspire.Hosting.Dapr.Tests.csproj:9]:
```suggestion
```

What's this used for?

File: src/Aspire.Hosting/Aspire.Hosting.csproj (hunk 3 above)
Comment (sebastienros) [reply src/Aspire.Hosting/Aspire.Hosting.csproj:71]: `await app.ExecuteBeforeStartHooksAsync(default);` in `DaprTests`

Comment (eerhardt) [reply src/Aspire.Hosting/Aspire.Hosting.csproj:71]: Can we refactor the tests to not need to use an internal API?

**Comment by sebastienros** [comment]
@radical 

Is it possible that the helix jobs don't have all the required files? This works locally.

```
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.3+1b45f5407b (64-bit .NET 8.0.7)
[xUnit.net 00:00:03.43]   Discovering: Aspire.Hosting.Dapr.Tests
[xUnit.net 00:00:03.47]   Discovered:  Aspire.Hosting.Dapr.Tests
[xUnit.net 00:00:03.47]   Starting:    Aspire.Hosting.Dapr.Tests
[xUnit.net 00:00:03.62]     Aspire.Hosting.Dapr.Tests.DaprSchemaTests.ValidateWithDaprManifest [FAIL]
[xUnit.net 00:00:03.62]       Aspire.Hosting.DistributedApplicationException : Project file 'D:\a\_work\1\s\tests\testproject\TestProject.ServiceA\TestProject.ServiceA.csproj' was not found.
[xUnit.net 00:00:03.62]       Stack Trace:
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/LaunchProfileExtensions.cs(66,0): at Aspire.Hosting.LaunchProfileExtensions.GetLaunchSettings(IProjectMetadata projectMetadata, String resourceName)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/LaunchProfileExtensions.cs(28,0): at Aspire.Hosting.LaunchProfileExtensions.GetLaunchSettings(ProjectResource projectResource)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/LaunchProfileExtensions.cs(106,0): at Aspire.Hosting.LaunchProfileExtensions.TrySelectLaunchProfileByOrder(ProjectResource projectResource, String& launchProfileName)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/LaunchProfileExtensions.cs(168,0): at Aspire.Hosting.LaunchProfileExtensions.SelectLaunchProfileName(ProjectResource projectResource)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/LaunchProfileExtensions.cs(33,0): at Aspire.Hosting.LaunchProfileExtensions.GetEffectiveLaunchProfile(ProjectResource projectResource, Boolean throwIfNotFound)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs(318,0): at Aspire.Hosting.ProjectResourceBuilderExtensions.WithProjectDefaults(IResourceBuilder`1 builder, ProjectResourceOptions options)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs(216,0): at Aspire.Hosting.ProjectResourceBuilderExtensions.AddProject[TProject](IDistributedApplicationBuilder builder, String name, Action`1 configure)
[xUnit.net 00:00:03.62]         /_/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs(63,0): at Aspire.Hosting.ProjectResourceBuilderExtensions.AddProject[TProject](IDistributedApplicationBuilder builder, String name)
[xUnit.net 00:00:03.62]         /_/tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs(24,0): at Aspire.Hosting.Dapr.Tests.DaprSchemaTests.<>c.<ValidateWithDaprManifest>b__0_0(IDistributedApplicationBuilder builder)
[xUnit.net 00:00:03.62]         /_/tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs(184,0): at Aspire.Hosting.Tests.Schema.SchemaTests.ValidateApplicationSamples(String testCaseName, Action`1 configurator)
[xUnit.net 00:00:03.62]         /_/tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs(14,0): at Aspire.Hosting.Dapr.Tests.DaprSchemaTests.ValidateWithDaprManifest()
[xUnit.net 00:00:03.62]            at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
[xUnit.net 00:00:03.62]            at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[xUnit.net 00:00:03.72]   Finished:    Aspire.Hosting.Dapr.Tests

```

File: src/Aspire.Hosting/Aspire.Hosting.csproj (hunk 3 above)
Comment (sebastienros) [reply src/Aspire.Hosting/Aspire.Hosting.csproj]: Using UnsafeAccessor instead

File: tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            {
+                var dapr = builder.AddDapr(dopts =>
+                {
+                    // Just to avoid dynamic discovery which will throw.
+                    dopts.DaprPath = "notrealpath";
+                });
+                var state = dapr.AddDaprStateStore("daprstate");
+                var pubsub = dapr.AddDaprPubSub("daprpubsub");
+
+                builder.AddProject<Projects.ServiceA>("project")
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Dapr.Tests/DaprSchemaTests.cs:24]: Use a mock Project like we do here:

https://github.com/dotnet/aspire/blob/7378176c2128f5cc26764a48e118212a0a3cbfe5/tests/Aspire.Hosting.Qdrant.Tests/AddQdrantTests.cs#L179-L180

File: tests/Aspire.Hosting.Dapr.Tests/Aspire.Hosting.Dapr.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <RunTestsOnHelix>false</RunTestsOnHelix>
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Dapr.Tests/Aspire.Hosting.Dapr.Tests.csproj:5]:
```suggestion
```
Can we remove this now?

**Comment by radical** [comment]
We should add https://github.com/dotnet/aspire/blob/main/tests/Aspire.Workload.Tests/xunit.runner.json + https://github.com/dotnet/aspire/blob/d431c2839b03430617d7d066593ea639ed559124/tests/Aspire.Workload.Tests/Aspire.Workload.Tests.csproj#L9 to the playground tests. That would should us which test is stuck, and for how long.

**Comment by radical** [comment]
It has ` "longRunningTestSeconds": 120,`.

**Comment by radical** [comment]
Fixed the conflicts.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**